Fix lifecycle rules not applied in some cases

vadmeste opened the pull request:
## Description
HasActiveRules was not behaving as expected, this commit fixes it
and adds more unit tests.

## Motivation and Context
Fixing lifecycle expiration in some cases


## How to test this PR?
1. Apply a lifecycle rule with prefix = "doc/"
2. Upload objects to "doc/"
3. Change MinIO code to accelerate crawling

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

klauspost reviewed (approved):
Thanks for the fix. LGTM!

vadmeste committed:
Fix lifecycle rules not applied in some cases

HasActiveRules was not behaving as expected, this commit fixes it
and adds more unit tests.
